pkg/util/lifted/objectwatcher.go

```diff
@@ -84,5 +84,10 @@ func objectMetaObjEquivalent(a, b metav1.Object) bool {
 	if !reflect.DeepEqual(aLabels, bLabels) && (len(aLabels) != 0 || len(bLabels) != 0) {
 		return false
 	}
+	aAnnotations := a.GetAnnotations()
+	bAnnotations := b.GetAnnotations()
+	if !reflect.DeepEqual(aAnnotations, bAnnotations) && (len(aAnnotations) != 0 || len(bAnnotations) != 0) {
+		return false
+	}
```

Comment:
This change may be affected by the logic. We can propose a PR separately and analyze the impact.

Reply:
> This change may be affected by the logic. We can propose a PR separately and analyze the impact.

OK,  see another PR in #3792